src/main/java/org/terasology/simpleFarming/components/SaplingDefinitionComponent.java

```diff
@@ -0,0 +1,77 @@
+package org.terasology.simpleFarming.components;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.simpleFarming.systems.TreeAuthoritySystem;
+import org.terasology.world.block.Block;
+
+/**
+ * Stores the necessary data for defining a tree.
+ * <p>
+ * In order to use it, add the component to a prefab, then link to that
+ * prefab from a {@link SeedDefinitionComponent}. Alternatively, the component
+ * can be included directly in the item and the SeedDefinitionComponent left
+ * empty.
+ * <p>
+ * After the sapling has been placed, this component is transferred to
+ * the sapling block's associated entity. When the sapling grows, the
+ * {@link RootComponent} of a tree is created based on the sapling's
+ * definition component.
+ */
+public class SaplingDefinitionComponent implements Component {
+	
+	/**
+	 * The location of the sapling. This does not need to be provided with the
+	 * definition, it is automatically set by {@link TreeAuthoritySystem}
+	 */
+	public Vector3i location;
+	
+	/**
+	 * The block to use for the sapling in the world.
+	 */
+	public Block sapling;
+	
+	/**
+	 * The block to use for a log in the tree in the world.
+	 */
+	public Block log;
```

Comment:
So, one other change I'd like to make would be to differentiate leaves from fruit bearing versions. At present, every leaf in the tree is a fruit bush. This results in things like growing a single tree fully and getting almost 200 saplings back. Unfortunately that's also not balance-able on the bush end, because you can't alter how many seeds are dropped.

Ideally there'd be a 'leaf' entry that would be used for most of the leaf blocks and then have a 'bush' entry that would be used to replace leaves with randomly. Would also be nice to have a field to control the percentage of leaves that get converted to bushes. (With a reasonable default as usual).

Again, not needed for GCI but needed eventually. If you have an alternate idea or think we should leave it as is then please let me know. It's your feature :)

Reply:
Would changing the Bush system to allow for changing drop chances in the component work as well? There could be a default value which makes it behave the way it does right now so older plants don't break.

Reply:
That should work yeah. If you had the default as 100% then it wouldn't change the others

You should also probably only make the percentage drop chance to only work in the instance where the bush only drops a single seed

Reply:
I made a system that uses weights for each amount of saplings dropped, e.g. [1, 2, 1] means 25% chance to drop 0, 50% chance to drop 1, 25% chance to drop 2. The default is [0, 1, 1, 1] which should work identically to the previous behavior. Leaves now use [36, 3, 1], so 90% chance to drop 0, 7.5% chance to drop 1, 2.5% chance to drop 2. This seems to produce pretty balanced results.